pandas/core/series.py

```diff
@@ -4997,12 +5002,24 @@ def between(self, left, right, inclusive=True) -> Series:
         3    False
         dtype: bool
         """
-        if inclusive:
+
+        if inclusive == "both" or inclusive is True:
```

Comment:
handle the bool values (the deprecated ones) as separate cases to emit a warning (and then simply translate them the correct inclusive)